src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -93,20 +93,25 @@ public static class AttributeMatcher
             var presentAttributes = new List<string>();
             var missingAttributes = new List<string>();
 
-            foreach (var attribute in requiredAttributes)
+            foreach (var requiredAttribute in requiredAttributes)
             {
-                if (!context.AllAttributes.ContainsName(attribute) ||
-                    context.AllAttributes[attribute] == null ||
-                    (context.AllAttributes[attribute].Value is string &&
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute].Value as string)))
+                IReadOnlyTagHelperAttribute attribute;
+                if (!context.AllAttributes.TryGetAttribute(requiredAttribute, out attribute))
                 {
-                    // Missing attribute!
-                    missingAttributes.Add(attribute);
+                    // Missing attribute.
+                    missingAttributes.Add(requiredAttribute);
+                    continue;
                 }
-                else
+
+                var valueAsString = attribute.Value as string;
+                if (valueAsString != null && string.IsNullOrEmpty(valueAsString))
                 {
-                    presentAttributes.Add(attribute);
+                    // Treat minimized attributes or attributes with empty values as missing.
```

Comment:
I misspoke earlier and this does not handle minimized attributes -- they're treated as present. This is just about an empty string value. Put another way, the only `true` case is `string.Equals(attribute.Value, string,Empty)`.